lib/resourceapply/apiext.go

```diff
@@ -3,17 +3,17 @@ package resourceapply
 import (
 	"context"
 
+	"github.com/google/go-cmp/cmp"
 	"github.com/openshift/cluster-version-operator/lib/resourcemerge"
 	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	apiextclientv1 "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/util/diff"
 	"k8s.io/klog/v2"
 	"k8s.io/utils/pointer"
 )
 
-func ApplyCustomResourceDefinitionv1(ctx context.Context, client apiextclientv1.CustomResourceDefinitionsGetter, required *apiextv1.CustomResourceDefinition) (*apiextv1.CustomResourceDefinition, bool, error) {
+func ApplyCustomResourceDefinitionv1(ctx context.Context, client apiextclientv1.CustomResourceDefinitionsGetter, required *apiextv1.CustomResourceDefinition, reconciling bool) (*apiextv1.CustomResourceDefinition, bool, error) {
```

Comment:
Can we pass in the mode instead of passing in a `reconciling` bool?  Like we do for ClusterOperators [here][1]?  That way we don't have to bump cluster signatures if we need to make some other mode distinction later on.

[1]: https://github.com/openshift/cluster-version-operator/blob/3b1047de4d2a82d2a1f4f92c224a1c46eba72ca9/pkg/cvo/internal/operatorstatus.go#L123-L126

Reply:
I considered that and decided against it because it would leak abstractions. `Mode` is part of the `resourcebuilder` package, where as the resource creation/update logic is in the `internal`. If in the future the full mode is required more refactoring can be done where the mode is moved to a common package and then it can be passed to the apply function.